internal/terminal/table_test.go

```diff
@@ -0,0 +1,263 @@
+package terminal
+
+import (
+	"fmt"
+	"github.com/10gen/realm-cli/internal/utils/test/assert"
+	"github.com/fatih/color"
+	"testing"
+)
+
+func TestEmptyTableConstructor(t *testing.T) {
+	for _, tc := range []struct {
+		description string
+		header      []string
+		data        []map[string]interface{}
+	}{
+		{
+			description: "Should return empty table if there is nil for the header",
+			header:      nil,
+			data:
+			[]map[string]interface{}{
+				{"test": "should not show up"},
+			},
+		},
+		{
+			description: "Should return empty table if there is no header information",
+			header:      []string{},
+			data: []map[string]interface{}{
+				{"test": "should not show up"},
+			},
+		},
+	} {
+		t.Run(tc.description, func(*testing.T) {
+			table := newTable(tc.header, tc.data)
+			assert.Nil(t, table.headers)
+			assert.Nil(t, table.data)
+			assert.Nil(t, table.columnWidths)
+		})
+	}
+}
+
+func TestTableConstructor(t *testing.T) {
+	t.Run("Should return a table with only headers if there is one and no data", func(t *testing.T) {
+		headers := []string{"should", "show", "up", "only"}
+		table := newTable(headers, nil)
+		assert.Equal(t, table.headers, headers)
+		assert.Equal(t, len(table.data), 0)
+		assert.Equal(t, table.columnWidths, map[string]int{
+			"should": 6,
+			"show":   4,
+			"up":     2,
+			"only":   4,
+		})
+	})
+	t.Run("Should return a table if there is both a header and data", func(t *testing.T) {
+		table := newTable(
+			[]string{"test"},
+			[]map[string]interface{}{
+				{
+					"test":   "should show up",
+					"int":    1,
+					"float":  1.000,
+					"float2": 1.234567890123,
+					"array":  []string{"1", "2", "3"},
+					"nil":    nil,
+				},
+			},
+		)
+		assert.Equal(t, table.headers, []string{"test"})
+		assert.Equal(t, table.data, []map[string]string{
+			{
+				"test":   "should show up",
+			},
+		})
+	})
+}
+
+func TestTableMessage(t *testing.T) {
+	t.Run("Should return an empty string and an error if the table has no headers", func(t *testing.T) {
```

Comment:
with enough test cases written for `TestNewTable`, then the test for `TestTableMessage` and `TestTablePayload` should end up being very simple, we'll likely only want to confirm:
1. with normal data, normal output is produced (edge case-y tests should likely all live in `TestNewTable`)
1. with `nil` headers, we return an error

I see that's basically what you already have with `TestTablePayload`, so lets just port the edge cases from `TestTableMessage` here to `TestNewTable`